apps/src/templates/teacherDashboard/TeacherDashboardHeader.jsx

```diff
@@ -47,7 +48,8 @@ function TeacherDashboardHeader({
   const inUSA =
     ['US', 'RD'].includes(currentUser.countryCode) || !!currentUser.usStateCode;
   const showAgeGatedStudentsBanner =
-    inUSA && currentUser.isTeacher && ageGatedStudentsCount > 0;
+    (inUSA && currentUser.isTeacher && ageGatedStudentsCount > 0) ||
+    DCDO.get('show-age-gated-students-banner', false);
```

Comment:
```suggestion
    (inUSA && currentUser.isTeacher && ageGatedStudentsCount > 0) &&
    DCDO.get('show-age-gated-students-banner', !Rails.env.production?);
```
This should be an `&&` otherwise the banner will always be shown once this DCDO flag is turned on.

I also added `!Rails.env.production?` so the flag is on by default for dev, staging, and test but turned off by default in production.